File: app/src/test/java/com/abishov/hexocat/commons/schedulers/TrampolineSchedulersProvider.java
package com.abishov.hexocat.commons.schedulers;

import io.reactivex.Scheduler;
import io.reactivex.schedulers.Schedulers;

public final class TrampolineSchedulersProvider implements SchedulerProvider {

    @Override
    public Scheduler computation() {
        return Schedulers.trampoline();
    }

    @Override
    public Scheduler io() {
        return Schedulers.trampoline();
    }

    @Override
    public Scheduler ui() {
        return Schedulers.trampoline();
    }
}

File: app/src/test/java/com/abishov/hexocat/common/schedulers/TrampolineSchedulersProvider.kt
package com.abishov.hexocat.common.schedulers

import io.reactivex.schedulers.Schedulers

class TrampolineSchedulersProvider : SchedulerProvider {

  override fun computation() = Schedulers.trampoline()

  override fun io() = Schedulers.trampoline()

  override fun ui() = Schedulers.trampoline()
}
